### PR TITLE
CDPAM-116 | Persisting public key cert to be used to configure knox t…

### PR DESCRIPTION
**Pull request by dvguruprasad**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java [hunk 1]
```diff
@@ -39,18 +52,31 @@ public String selector() {
...
+                return new ClusterProxyRegistrationSuccess(request.getResourceId());
             }
+            ConfigRegistrationResponse registerResponse = clusterProxyService.registerCluster(stack);
+            Cluster cluster = stack.getCluster();
+            if (cluster.hasGateway()) {
+                LOGGER.debug("Updating Gateway for cluster {} in environment {} with public key certificate retrieved from Cluster Proxy",
+                        cluster.getId(), stack.getEnvironmentCrn());
+                Gateway gateway = cluster.getGateway();
+                gateway.setTokenCert(registerResponse.getX509Unwrapped());
+                gateway.setSsoType(SSOType.SSO_PROVIDER);
```
Comment (keyki) [inline core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java]: Do we need to update the sso type? I think this comes from the API requests.

Comment (dvguruprasad) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java]: When debugging locally we didn't see it getting populated - gets persisted as 'NONE'. Was this changed recently?

Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java]: NONE is the default value if you don't provide it on the API. 
```
    "gateway": {
      "topologies": [
        {
          "topologyName": "dp-proxy",
          "exposedServices": [
            "ALL"
          ]
        }
      ],
      "ssoType": "SSO_PROVIDER"
    }
```
SDX and DistroX already sets them to `SSO_PROVIDER`

Comment (dvguruprasad) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java]: Ok. Looks like this is not getting set from the UI.

**Pull request merged**